### PR TITLE
Site Settings: Show an upgrade banner in Antispam card for free plan

### LINKED ISSUE #14654: Jetpack Settings: Antispam card looks odd with a Free plan
I've just connected a Jetpack site on a Free plan and went to the Security settings tab, but the Antispam card looks like this:

I think this could be confusing to the user. Should we add some kind of message here, or maybe a nudge to upgrade to a paid JP plan?

/cc @oskosk as he worked on the card and @rickybanister @MichaelArestad for ideas

### PR DESCRIPTION
**Pull request by tyxla**
This PR updates the Antispam card to show an upgrade banner when the user is on a free plan and does not have Akismet setup.

Fixes #14654.

I've used the copy from the plans, but I can still use some copy feedback, maybe folks from @Automattic/editorial can have a look.

Preview:

To test:
* Checkout this branch, or get it going on calypso.live
* Go to `/settings/security/:site`, where `:site` is a Jetpack site on a Free plan.
* Verify you can see the banner, and it leads to the plans page with the Antispam feature highlighted.
* Test with a Jetpack site with a paid plan and verify it works like it did before.

### CODE REVIEW

File: client/my-sites/site-settings/spam-filtering-settings.jsx [hunk 1]
```diff
@@ -40,6 +47,19 @@ const SpamFilteringSettings = ( {
 		( wordpress_api_key && isDirty && isStoredKey && ! hasAkismetKeyError );
 	const isInvalidKey = isDirty && hasAkismetKeyError && ! isStoredKey;
 	let validationText, className, header = null;
+
+	if ( ! inTransition && ! hasAkismetFeature && ! isValidKey ) {
+		return (
+			<Banner
+				description={ translate( 'Add state-of-the-art spam defense powered by Akismet.' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/spam-filtering-settings.jsx]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'State-of-the-art spam defense powered by Akismet.' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by rickybanister** [comment]
The banner looks good to me. We will need to address the other circumstances at a later date to give folks the chance to manually configure Akismet, but this help the majority for now.

**Comment by tyxla** [comment]
Thanks @rickybanister. Yeah, I've opened #16793 just to make sure we keep track of these and address them accordingly when the time comes.

**Comment by kristastevens** [comment]
Hello @tyxla! Thanks for the ping. How about a slightly stronger benefit in that first line and a short explanation of what Akismet does, specifically -- let me know what you think. 

Defend your site against spam! Upgrade to Jetpack Personal.
Detect and tweeze spam automatically, with Akismet.

**Comment by tyxla** [comment]
Thanks @kristastevens, I love your suggestion - it's much more engaging! 

Here's an updated preview:

**Comment by kristastevens** [comment]
Thank you @tyxla! Happy to help out!

**Comment by sirreal** [comment]
Not related, but I noticed in this PR. If we see the key input, it means we have a payed plan, which means we have a purchase. Is that correct?

In that case, should we consider directing the user to `/me/purchases` to retrieve their Akismet key, or can we do something smarter like just trying to fill it in, or providing a button to "try to find my key"? It seems silly to send them elsewhere to get a key, when we know it in another part of the application.

api-key

☝️  The link goes to https://akismet.com/wordpress/

We know the keys associated with a purchase:

keys

cc @rickybanister

**Comment by rickybanister** [comment]
That's very true @sirreal. I would much rather keep them within WordPress.com.

It is certainly possible they have an Akismet key from before, which we don't prevent them from using, but it's likely more helpful to keep them here. 

Another side issue is that we don't display non-plan purchases at /me/purchases so Akismet keys purchased on their own would not appear there.

cc @beaulebens just to keep you in this loop

**Comment by tyxla** [comment]
Thanks for the review and for the tips, folks! I've added your thoughts to #16793, so we make sure to implement these improvements when we get to that.

**Pull request merged**